src/transformers/modeling_encoder_decoder.py

```diff
@@ -108,129 +118,116 @@ def from_pretrained(
-                    + " the output directory. The default behaviour is to overwrite them."
-                )
-
-            # Empty the output directory
-            for directory_to_remove in sub_directories:
-                # Remove all files into the subdirectory
-                files_to_remove = os.listdir(os.path.join(save_directory, directory_to_remove))
-                for file_to_remove in files_to_remove:
-                    os.remove(os.path.join(save_directory, directory_to_remove, file_to_remove))
-                # Remove the subdirectory itself
-                os.rmdir(os.path.join(save_directory, directory_to_remove))
-
-            assert len(os.listdir(save_directory)) == 0  # sanity check
-
-        # Create the "encoder" directory inside the output directory and save the encoder into it
-        if not os.path.exists(os.path.join(save_directory, "encoder")):
-            os.mkdir(os.path.join(save_directory, "encoder"))
-        self.encoder.save_pretrained(os.path.join(save_directory, "encoder"))
-
-        # Create the "encoder" directory inside the output directory and save the decoder into it
-        if not os.path.exists(os.path.join(save_directory, "decoder")):
-            os.mkdir(os.path.join(save_directory, "decoder"))
-        self.decoder.save_pretrained(os.path.join(save_directory, "decoder"))
-
-    def forward(self, encoder_input_ids, decoder_input_ids, **kwargs):
-        """ The forward pass on a seq2eq depends what we are performing:
-
-        - During training we perform one forward pass through both the encoder
-          and decoder;
-        - During prediction, we perform one forward pass through the encoder,
-          and then perform several forward passes with the encoder's hidden
-          state through the decoder to decode a full sequence.
-
-        Therefore, we skip the forward pass on the encoder if an argument named
-        `encoder_hidden_state` is passed to this function.
-
         Params:
             encoder_input_ids: ``torch.LongTensor`` of shape ``(batch_size, sequence_length)``
                 Indices of encoder input sequence tokens in the vocabulary.
             decoder_input_ids: ``torch.LongTensor`` of shape ``(batch_size, sequence_length)``
                 Indices of decoder input sequence tokens in the vocabulary.
             kwargs: (`optional`) Remaining dictionary of keyword arguments.
         """
-        kwargs_encoder, kwargs_decoder = self.prepare_model_kwargs(**kwargs)
 
-        # Encode if needed (training, first prediction pass)
-        encoder_hidden_states = kwargs_encoder.pop("hidden_states", None)
-        if encoder_hidden_states is None:
-            encoder_outputs = self.encoder(encoder_input_ids, **kwargs_encoder)
-            encoder_hidden_states = encoder_outputs[0]
-        else:
-            encoder_outputs = ()
+        kwargs_encoder = {argument: value for argument, value in kwargs.items() if not argument.startswith("decoder_")}
```

Comment:
so neutral kwargs like use_cache only go to encoder? Is that desired/irrelevant?

Reply:
I think the user will always have to prepend each argument with `decoder_` in order to use it for the encoder. So if one would use a Bert + GPT2 (which is not supported at the moment), one would have to write `decoder_use_cache=True`. IMO this is ok.